### PR TITLE
Validate component kind

### PR DESCRIPTION
**Pull request by Ghostbaby**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ IMG ?= controller:latest
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
 # apiextensions.k8s.io/v1beta1 are Deprecated in v1.16 in favor of apiextensions.k8s.io/v1
 # ref https://github.com/kubernetes-sigs/kubebuilder/issues/1065
-CRD_OPTIONS ?= "crd:trivialVersions=true,crdVersions=v1"
```
Comment (soulseen) [inline Makefile:7]: why remove `crdVersions=v1`? `v1beta1` is deprecated in v1.16 in favor of apiextensions.k8s.io/v1

**Comment by soulseen** [comment]
looks good ! little changes otherwise lgtm.

File: Makefile (hunk 1 above)
Comment (Ghostbaby) [reply Makefile:7]: Maybe it's because I use the latest version of kubebuilder to generate the webhook.😂

File: api/v1/harborcluster_webhook.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+
+func (r *HarborCluster) ValidateComponentKind(old runtime.Object) error {
+	oldHarbor := old.(*HarborCluster)
+	if r.Spec.Redis.Kind != oldHarbor.Spec.Redis.Kind ||
+		r.Spec.Database.Kind != oldHarbor.Spec.Database.Kind ||
+		r.Spec.Storage.Kind != oldHarbor.Spec.Storage.Kind {
+		return errors.New("component type switching is not supported")
```
Comment (steven-zou) [inline api/v1/harborcluster_webhook.go:81]:
```suggestion
		return errors.New("service kind switching is not supported")
```

File: main.go [hunk 3]
```diff
@@ -17,10 +17,11 @@ package main
 
 import (
 	"flag"
-	redisCli "github.com/spotahome/redis-operator/api/redisfailover/v1"
 	"os"
 	"time"
 
+	redisCli "github.com/spotahome/redis-operator/api/redisfailover/v1"
+
```
Comment (steven-zou) [inline main.go:24]:
```suggestion
```

**Pull request merged**